### ISSUE TITLE
System.Diagnostics - Undoc 6.0 APIs

### ISSUE THREAD

**Issue by carlossanlop** [dotnet/runtime#58654]
The following APIs were introduced in 6.0 and did not get documented in triple slash. They will need to be manually documented directly in dotnet-api-docs, and must target the net6-rc1 branch. This needs to be done before Tuesday September 14th so we have enough time to generate the IntelliSense package that will be shipped to GA.

| Undoc APIs                                                                                                                                                                                                                                                                                                           |
|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| [M:System.Diagnostics.Debug.Assert(System.Boolean,System.Diagnostics.Debug.AssertInterpolatedStringHandler@)](https://github.com/dotnet/dotnet-api-docs/blob/net6-rc1/xml/System.Diagnostics/Debug.xml)                                                                                                              |
| [M:System.Diagnostics.Debug.Assert(System.Boolean,System.Diagnostics.Debug.AssertInterpolatedStringHandler@,System.Diagnostics.Debug.AssertInterpolatedStringHandler@)](https://github.com/dotnet/dotnet-api-docs/blob/net6-rc1/xml/System.Diagnostics/Debug.xml)                                                    |
| [M:System.Diagnostics.Debug.WriteIf(System.Boolean,System.Diagnostics.Debug.WriteIfInterpolatedStringHandler@)](https://github.com/dotnet/dotnet-api-docs/blob/net6-rc1/xml/System.Diagnostics/Debug.xml)                                                                                                            |
| [M:System.Diagnostics.Debug.WriteIf(System.Boolean,System.Diagnostics.Debug.WriteIfInterpolatedStringHandler@,System.String)](https://github.com/dotnet/dotnet-api-docs/blob/net6-rc1/xml/System.Diagnostics/Debug.xml)                                                                                              |
| [M:System.Diagnostics.Debug.WriteLineIf(System.Boolean,System.Diagnostics.Debug.WriteIfInterpolatedStringHandler@,System.String)](https://github.com/dotnet/dotnet-api-docs/blob/net6-rc1/xml/System.Diagnostics/Debug.xml)                                                                                          |
| [M:System.Diagnostics.Debug.WriteLineIf(System.Boolean,System.Diagnostics.Debug.WriteIfInterpolatedStringHandler@)](https://github.com/dotnet/dotnet-api-docs/blob/net6-rc1/xml/System.Diagnostics/Debug.xml)                                                                                                        |
| [M:System.Diagnostics.Debug.AssertInterpolatedStringHandler.AppendFormatted\`\`1(\`\`0,System.String)](https://github.com/dotnet/dotnet-api-docs/blob/net6-rc1/xml/System.Diagnostics/Debug+AssertInterpolatedStringHandler.xml)                                                                                         |
| [M:System.Diagnostics.Debug.AssertInterpolatedStringHandler.AppendFormatted\`\`1(\`\`0,System.Int32)](https://github.com/dotnet/dotnet-api-docs/blob/net6-rc1/xml/System.Diagnostics/Debug+AssertInterpolatedStringHandler.xml)                                                                                          |
| [M:System.Diagnostics.Debug.AssertInterpolatedStringHandler.AppendFormatted\`\`1(\`\`0,System.Int32,System.String)](https://github.com/dotnet/dotnet-api-docs/blob/net6-rc1/xml/System.Diagnostics/Debug+AssertInterpolatedStringHandler.xml)                                                                            |
| [M:System.Diagnostics.Debug.AssertInterpolatedStringHandler.AppendFormatted\`\`1(\`\`0)](https://github.com/dotnet/dotnet-api-docs/blob/net6-rc1/xml/System.Diagnostics/Debug+AssertInterpolatedStringHandler.xml)                                                                                                       |
| [M:System.Diagnostics.Debug.WriteIfInterpolatedStringHandler.AppendFormatted\`\`1(\`\`0,System.Int32,System.String)](https://github.com/dotnet/dotnet-api-docs/blob/net6-rc1/xml/System.Diagnostics/Debug+WriteIfInterpolatedStringHandler.xml)                                                                          |
| [M:System.Diagnostics.Debug.WriteIfInterpolatedStringHandler.AppendFormatted\`\`1(\`\`0,System.String)](https://github.com/dotnet/dotnet-api-docs/blob/net6-rc1/xml/System.Diagnostics/Debug+WriteIfInterpolatedStringHandler.xml)                                                                                       |
| [M:System.Diagnostics.Debug.WriteIfInterpolatedStringHandler.AppendFormatted\`\`1(\`\`0,System.Int32)](https://github.com/dotnet/dotnet-api-docs/blob/net6-rc1/xml/System.Diagnostics/Debug+WriteIfInterpolatedStringHandler.xml)                                                                                        |
| [M:System.Diagnostics.Debug.WriteIfInterpolatedStringHandler.AppendFormatted\`\`1(\`\`0)](https://github.com/dotnet/dotnet-api-docs/blob/net6-rc1/xml/System.Diagnostics/Debug+WriteIfInterpolatedStringHandler.xml)                                                                                                     |
| [T:System.Diagnostics.DiagnosticListenerExtensions](https://github.com/dotnet/dotnet-api-docs/blob/net6-rc1/xml/System.Diagnostics/DiagnosticListenerExtensions.xml)                                                                                                                                                 |
| [M:System.Diagnostics.DiagnosticListenerExtensions.SubscribeWithAdapter(System.Diagnostics.DiagnosticListener,System.Object)](https://github.com/dotnet/dotnet-api-docs/blob/net6-rc1/xml/System.Diagnostics/DiagnosticListenerExtensions.xml)                                                                       |
| [M:System.Diagnostics.DiagnosticListenerExtensions.SubscribeWithAdapter(System.Diagnostics.DiagnosticListener,System.Object,System.Func{System.String,System.Object,System.Object,System.Boolean})](https://github.com/dotnet/dotnet-api-docs/blob/net6-rc1/xml/System.Diagnostics/DiagnosticListenerExtensions.xml) |
| [M:System.Diagnostics.DiagnosticListenerExtensions.SubscribeWithAdapter(System.Diagnostics.DiagnosticListener,System.Object,System.Func{System.String,System.Boolean})](https://github.com/dotnet/dotnet-api-docs/blob/net6-rc1/xml/System.Diagnostics/DiagnosticListenerExtensions.xml)                             |
| [M:System.Diagnostics.DistributedContextPropagator.#ctor](https://github.com/dotnet/dotnet-api-docs/blob/net6-rc1/xml/System.Diagnostics/DistributedContextPropagator.xml)                                                                                                                                           |
| [M:System.Diagnostics.DistributedContextPropagator.CreateDefaultPropagator](https://github.com/dotnet/dotnet-api-docs/blob/net6-rc1/xml/System.Diagnostics/DistributedContextPropagator.xml)                                                                                                                         |
| [M:System.Diagnostics.DistributedContextPropagator.CreateNoOutputPropagator](https://github.com/dotnet/dotnet-api-docs/blob/net6-rc1/xml/System.Diagnostics/DistributedContextPropagator.xml)                                                                                                                        |
| [M:System.Diagnostics.DistributedContextPropagator.CreatePassThroughPropagator](https://github.com/dotnet/dotnet-api-docs/blob/net6-rc1/xml/System.Diagnostics/DistributedContextPropagator.xml)                                                                                                                     |
| [N:System.Diagnostics.Metrics](https://github.com/dotnet/dotnet-api-docs/blob/net6-rc1/xml/ns-System.Diagnostics.Metrics.xml)                                                                                                                                                                                        |
| [T:System.Diagnostics.Metrics.Counter\`1](https://github.com/dotnet/dotnet-api-docs/blob/net6-rc1/xml/System.Diagnostics.Metrics/Counter\`1.xml)                                                                                                                                                                       |
| [T:System.Diagnostics.Metrics.Histogram\`1](https://github.com/dotnet/dotnet-api-docs/blob/net6-rc1/xml/System.Diagnostics.Metrics/Histogram\`1.xml)                                                                                                                                                                   |
| [T:System.Diagnostics.Metrics.Instrument\`1](https://github.com/dotnet/dotnet-api-docs/blob/net6-rc1/xml/System.Diagnostics.Metrics/Instrument\`1.xml)                                                                                                                                                                 |
| [T:System.Diagnostics.Metrics.Measurement\`1](https://github.com/dotnet/dotnet-api-docs/blob/net6-rc1/xml/System.Diagnostics.Metrics/Measurement\`1.xml)                                                                                                                                                               |
| [T:System.Diagnostics.Metrics.MeasurementCallback\`1](https://github.com/dotnet/dotnet-api-docs/blob/net6-rc1/xml/System.Diagnostics.Metrics/MeasurementCallback\`1.xml)                                                                                                                                               |
| [M:System.Diagnostics.Metrics.Meter.CreateCounter\`\`1(System.String,System.String,System.String)](https://github.com/dotnet/dotnet-api-docs/blob/net6-rc1/xml/System.Diagnostics.Metrics/Meter.xml)                                                                                                                   |
| [M:System.Diagnostics.Metrics.Meter.CreateHistogram\`\`1(System.String,System.String,System.String)](https://github.com/dotnet/dotnet-api-docs/blob/net6-rc1/xml/System.Diagnostics.Metrics/Meter.xml)                                                                                                                 |
| [M:System.Diagnostics.Metrics.Meter.CreateObservableCounter\`\`1(System.String,System.Func{\`\`0},System.String,System.String)](https://github.com/dotnet/dotnet-api-docs/blob/net6-rc1/xml/System.Diagnostics.Metrics/Meter.xml)                                                                                        |
| [M:System.Diagnostics.Metrics.Meter.CreateObservableCounter\`\`1(System.String,System.Func{System.Diagnostics.Metrics.Measurement{\`\`0}},System.String,System.String)](https://github.com/dotnet/dotnet-api-docs/blob/net6-rc1/xml/System.Diagnostics.Metrics/Meter.xml)                                                |
| [M:System.Diagnostics.Metrics.Meter.CreateObservableCounter\`\`1(System.String,System.Func{System.Collections.Generic.IEnumerable{System.Diagnostics.Metrics.Measurement{\`\`0}}},System.String,System.String)](https://github.com/dotnet/dotnet-api-docs/blob/net6-rc1/xml/System.Diagnostics.Metrics/Meter.xml)        |
| [M:System.Diagnostics.Metrics.Meter.CreateObservableGauge\`\`1(System.String,System.Func{System.Collections.Generic.IEnumerable{System.Diagnostics.Metrics.Measurement{\`\`0}}},System.String,System.String)](https://github.com/dotnet/dotnet-api-docs/blob/net6-rc1/xml/System.Diagnostics.Metrics/Meter.xml)          |
| [M:System.Diagnostics.Metrics.Meter.CreateObservableGauge\`\`1(System.String,System.Func{System.Diagnostics.Metrics.Measurement{\`\`0}},System.String,System.String)](https://github.com/dotnet/dotnet-api-docs/blob/net6-rc1/xml/System.Diagnostics.Metrics/Meter.xml)                                                  |
| [M:System.Diagnostics.Metrics.Meter.CreateObservableGauge\`\`1(System.String,System.Func{\`\`0},System.String,System.String)](https://github.com/dotnet/dotnet-api-docs/blob/net6-rc1/xml/System.Diagnostics.Metrics/Meter.xml)                                                                                          |
| [M:System.Diagnostics.Metrics.MeterListener.SetMeasurementEventCallback\`\`1(System.Diagnostics.Metrics.MeasurementCallback{\`\`0})](https://github.com/dotnet/dotnet-api-docs/blob/net6-rc1/xml/System.Diagnostics.Metrics/MeterListener.xml)                                                                           |
| [T:System.Diagnostics.Metrics.ObservableCounter\`1](https://github.com/dotnet/dotnet-api-docs/blob/net6-rc1/xml/System.Diagnostics.Metrics/ObservableCounter\`1.xml)                                                                                                                                                   |
| [T:System.Diagnostics.Metrics.ObservableGauge\`1](https://github.com/dotnet/dotnet-api-docs/blob/net6-rc1/xml/System.Diagnostics.Metrics/ObservableGauge\`1.xml)                                                                                                                                                       |
| [T:System.Diagnostics.Metrics.ObservableInstrument\`1](https://github.com/dotnet/dotnet-api-docs/blob/net6-rc1/xml/System.Diagnostics.Metrics/ObservableInstrument\`1.xml)                                                                                                                                             |
| [M:System.Diagnostics.Metrics.ObservableInstrument\`1.Observe](https://github.com/dotnet/dotnet-api-docs/blob/net6-rc1/xml/System.Diagnostics.Metrics/ObservableInstrument\`1.xml)                                                                                                                                     |

**Comment by tarekgh** [comment]
@carlossanlop 

> The following APIs were introduced in 6.0 and did not get documented in triple slash. 

Are you sure? 😊

https://github.com/dotnet/runtime/blob/main/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DistributedContextPropagator.cs
https://github.com/dotnet/runtime/blob/main/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/Meter.cs
https://github.com/dotnet/runtime/blob/0becc9905f69968a2c15ba5d41a9ea8162517590/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/MeterListener.cs#L118
https://github.com/dotnet/runtime/blob/main/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/ObservableCounter.cs
https://github.com/dotnet/runtime/blob/main/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/ObservableGauge.cs
https://github.com/dotnet/runtime/blob/main/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/ObservableInstrument.cs

**Comment by carlossanlop** [comment]
Yep, I',m sure. The examples you listed are missing triple slash docs, so there was nothing to port, and need the documentation to be added manually. For example:

#### DistributedContextPropagator

- One of the undoc APIs in this type is the `CreateNoOutputPropagator` method, which only has a summary, but does not have a returns value: [runtime/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DistributedContextPropagator.cs#L105](https://github.com/dotnet/runtime/blob/1fd3222cd3ef71708405c2642b5e68cc5053170c/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DistributedContextPropagator.cs#L105)
- The dotnet-api-docs xml: [dotnet-api-docs/xml/System.Diagnostics/DistributedContextPropagator.xml#L90](https://github.com/dotnet/dotnet-api-docs/blob/584ba10cee2694dde6c81e66247ab2a160dc69d7/xml/System.Diagnostics/DistributedContextPropagator.xml#L90)

#### ObservableCounter

- One of the undoc APIs in this type is the class type parameter `T`, which is not documented on top of the class name: [runtime/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/ObservableCounter.cs#L19](https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/ObservableCounter.cs#L19)
- The dotnet-api-docs xml: [runtime/xml/System.Diagnostics.Metrics/ObservableCounter%601.xml#L29](https://github.com/dotnet/dotnet-api-docs/blob/b883cb36b3b69306881f1ea3fa70cf01ecb31439/xml/System.Diagnostics.Metrics/ObservableCounter%601.xml#L29)

The others are similar, @tarekgh . Just look for `To be added.` in their dotnet-api-docs file and you'll find which API is missing docs.

**Comment by carlossanlop** [comment]
@noahfalk @tommcdon can you please take care of these APIs? @tarekgh is going to be OOF until after the deadline for merging 6.0 docs.

**Comment by josalem** [comment]
@carlossanlop, @noahfalk is similarly OOF till after that deadline. I'll attempt to fill in the blanks, but I can't promise that I'm familiar enough with all the APIs to add too much detail.

Based on context, I'm assuming I can `ctrl`+`f` for "To be added." to find all the blanks I need to fill in. I'm assuming it's also preferential to fix up the `///` docs in the code as well?

**Comment by tommcdon** [comment]
@carlossanlop I'm taking ownership to drive this to completion

**Comment by carlossanlop** [comment]
@josalem that's correct, just fill the `To be added.` you find in those files. The only ones that are optional are the `<remarks>` items. Leave them untouched (do not remove the `To be added.`).

Thanks both. @tommcdon if you're going to work on them, please first try using [DocsPortingTool](https://github.com/carlossanlop/DocsPortingTool). Hopefully it will save some time.

**Comment by carlossanlop** [comment]
@tommcdon @josalem friendly reminder that today is the last day to merge docs for 6.0 APIs into the dotnet-api-docs net6-rc1 branch.

**Comment by carlossanlop** [comment]
I updated the list above to remove a few I found already existing in triple slash.

**Comment by tommcdon** [comment]
Closing as all PR's have been merged into RC1
